src/main/java/core/basesyntax/suppliers/FigureSupplier.java

```diff
@@ -0,0 +1,88 @@
+package core.basesyntax.suppliers;
+
+import core.basesyntax.Circle;
+import core.basesyntax.Figure;
+import core.basesyntax.IsoscelesTrapezoid;
+import core.basesyntax.Rectangle;
+import core.basesyntax.RightTriangle;
+import core.basesyntax.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int MAX_ARRAY_LENGTH = 100;
+    public static final double MIN_FIGURE_VALUE = 1;
+    public static final double MAX_FIGURE_VALUE = 1000;
+    public static final int NUMBER_OF_FIGURES = 5;
+
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure[] generateFigures() {
+        Figure[] randomFigures = new Figure[random.nextInt(MAX_ARRAY_LENGTH)];
+
+        for (int i = 0; i < randomFigures.length; i++) {
+
+            switch (random.nextInt(NUMBER_OF_FIGURES)) {
+                case 0:
+                    randomFigures[i] = getNewRandomSquare();
+                    break;
+                case 1:
+                    randomFigures[i] = getNewRandomRectangle();
+                    break;
+                case 2:
+                    randomFigures[i] = getNewRandomRightTriangle();
+                    break;
+                case 3:
+                    randomFigures[i] = getNewRandomCircle();
+                    break;
+                case 4:
+                    randomFigures[i] = getNewRandomIsoscelesTrapezoid();
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        return randomFigures;
+    }
+
+    public Square getNewRandomSquare() {
```

Comment:
should these methods be public?

Reply:
I thought that it would be nice to have a more robust code, in case I need to generate some random figure separately instead of using an array